deepchecks/vision/dataset.py

```diff
@@ -0,0 +1,140 @@
+from copy import copy
+from enum import Enum
+from collections import Counter
+from typing import Callable
+
+from torch.utils.data import DataLoader
+from torch import cat
+import logging
+
+from deepchecks.core.errors import DeepchecksValueError
+
+logger = logging.getLogger('deepchecks')
+
+__all__ = ['TaskType', 'VisionDataset']
+
+
+class TaskType(Enum):
+    """Enum containing supported task types."""
+
+    CLASSIFICATION = 'classification'
+    OBJECT_DETECTION = 'object_detection'
+    SEMANTIC_SEGMENTATION = 'semantic_segmentation'
+
+
+class VisionDataset:
+
+    _data: DataLoader = None
+
+    def __init__(self, data_loader: DataLoader, num_classes: int = None, label_type: str = None,
+                 label_transformer: Callable = None):
+        self._data = data_loader
+
+        if label_transformer is None:
+            self.label_transformer = lambda x: x
+        else:
+            self.label_transformer = label_transformer
+
+        if label_type is not None:
+            self.label_type = label_type
+        else:
+            self.label_type = self.infer_label_type()
+
+        self._num_classes = num_classes  # if not initialized, then initialized later in get_num_classes()
+        self._samples_per_class = None
+
+    def get_num_classes(self):
+        if self._num_classes is None:
+            samples_per_class = self.get_samples_per_class()
+            num_classes = len(samples_per_class.keys())
+            self._num_classes = num_classes
+        return self._num_classes
+
+    def get_samples_per_class(self):
+        if self._samples_per_class is None:
+            if self.label_type == TaskType.CLASSIFICATION.value:
+                counter = Counter()
+                for _ in range(len(self._data)):
+                    counter.update(self.label_transformer(next(iter(self._data))[1]))
```

Comment:
If anyone has a good idea of how to make sure that next(iter(self._data))[1] always gets self.label_transformer, it will be a welcome addition.

Reply:
Could be resolved as part of #758